source/common/router/router.h

```diff
@@ -573,7 +573,8 @@ class Filter : Logger::Loggable<Logger::Id::router>,
                                                      const Http::HeaderMap& headers) const;
 
   void maybeDoShadowing();
-  bool maybeRetryReset(Http::StreamResetReason reset_reason, UpstreamRequest& upstream_request);
+  bool maybeRetryReset(Http::StreamResetReason reset_reason, UpstreamRequest& upstream_request,
+                       bool is_timeout_retry);
```

Comment:
For readability, instead of using a bool and having the comments in various places I would recommend creating a 2 state enum. TimeoutRetry::Yes, TimeoutRetry::No. Same below.

Reply:
updated!